web/src/enterprise/campaigns/detail/patches/PatchNode.tsx

```diff
@@ -0,0 +1,138 @@
+import * as H from 'history'
+import { IPatch } from '../../../../../../shared/src/graphql/schema'
+import Octicon, { Diff } from '@primer/octicons-react'
+import React, { useState, useEffect, useCallback } from 'react'
+import { Link } from '../../../../../../shared/src/components/Link'
+import { ThemeProps } from '../../../../../../shared/src/theme'
+import { Collapsible } from '../../../../components/Collapsible'
+import { DiffStat } from '../../../../components/diff/DiffStat'
+import { FileDiffNode } from '../../../../components/diff/FileDiffNode'
+import { publishChangeset as _publishChangeset, queryPatchFileDiffs } from '../backend'
+import { LoadingSpinner } from '@sourcegraph/react-loading-spinner'
+import { Subject } from 'rxjs'
+import ErrorIcon from 'mdi-react/ErrorIcon'
+import { asError } from '../../../../../../shared/src/util/errors'
+import classNames from 'classnames'
+import { FileDiffConnection } from '../../../../components/diff/FileDiffConnection'
+import { FilteredConnectionQueryArgs } from '../../../../components/FilteredConnection'
+
+export interface PatchNodeProps extends ThemeProps {
+    node: IPatch
+    campaignUpdates?: Subject<void>
```

Comment:
To make clear this is not for subscribing
```suggestion
    campaignUpdates?: Pick<Observer<void>, 'next'>
```